RequestManagementSystem/Client/Request.py

```diff
@@ -527,14 +527,60 @@ def optimize( self ):
     # Set to True if the request could be optimized
     optimized = False
 
+    # Recognise Failover request series
+    repAndRegList = []
+    removeRepList = []
+    i = 0
+    while i < len( self.__operations__ ) - 1:
+      op1 = self.__operations__[i]
+      op2 = self.__operations__[i + 1]
+      # print i, getattr( op1, 'Type' ), getattr( op2, 'Type' )
+      if getattr( op1, 'Type' ) == 'ReplicateAndRegister' and \
+         getattr( op2, 'Type' ) == 'RemoveReplica':
+        fileSetA = set( list( f.LFN for f in op1 ) )
+        fileSetB = set( list( f.LFN for f in op2 ) )
+        if fileSetA == fileSetB:
+          # Source is useless if failover
+          if 'FAILOVER' in op1.SourceSE:
+            op1.SourceSE = ''
+          repAndRegList.append( ( op1.TargetSE, op1 ) )
+          removeRepList.append( ( op2.TargetSE, op2 ) )
+          del self.__operations__[i]
+          del self.__operations__[i]
```

Comment:
i+1 ?
This is a horrible fix based on the SE name which is completely arbitrary. If this optimization is necessary we should find a better way to identify the Failover SE or ( better ) the failover request setting logic in the job wrapper should be updated